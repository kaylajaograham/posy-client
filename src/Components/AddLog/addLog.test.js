import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddLogForm from './addLog';

describe(`addLog component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  };

  it('renders a form.addLog by default', () => {
    const wrapper = shallow(<AddLogForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('renders the addLog given props', () => {
    const wrapper = shallow(<AddLogForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})