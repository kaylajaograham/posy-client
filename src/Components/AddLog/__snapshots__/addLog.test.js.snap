// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addLog component renders a form.addLog by default 1`] = `
<form
  className="AddLogForm"
  onSubmit={[Function]}
>
  <div
    className="addLog"
  >
    <h3
      className="newLog"
    >
      New Log
    </h3>
    <div
      className="log_name"
    >
      <label
        htmlFor="addLogForm__log_name"
      >
        Title
      </label>
      <input
        id="addLogForm__log_name"
        name="log_name"
        onChange={[Function]}
      />
    </div>
    <div
      className="log_entry"
    >
      <label
        htmlFor="addLogForm__log_entry"
      >
        Entry
      </label>
      <textarea
        id="addLogForm__log_entry"
        name="log_entry"
        onChange={[Function]}
        type="text"
      />
    </div>
    <button
      className="button"
      type="submit"
    >
      Save
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </div>
</form>
`;

exports[`addLog component renders the addLog given props 1`] = `
<form
  className="AddLogForm"
  onSubmit={[Function]}
>
  <div
    className="addLog"
  >
    <h3
      className="newLog"
    >
      New Log
    </h3>
    <div
      className="log_name"
    >
      <label
        htmlFor="addLogForm__log_name"
      >
        Title
      </label>
      <input
        id="addLogForm__log_name"
        name="log_name"
        onChange={[Function]}
      />
    </div>
    <div
      className="log_entry"
    >
      <label
        htmlFor="addLogForm__log_entry"
      >
        Entry
      </label>
      <textarea
        id="addLogForm__log_entry"
        name="log_entry"
        onChange={[Function]}
        type="text"
      />
    </div>
    <button
      className="button"
      type="submit"
    >
      Save
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </div>
</form>
`;
